### PR TITLE
623 tsv helper function

### PR DESCRIPTION
**Pull request by exbotanical**

### CODE REVIEW

**Comment by effigies** [comment]
Ref: #623

File: bids-validator/validators/tsv/tsvParser.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * TSV
+ * Module for parsing TSV (and eventually other formats)
+ */
+
+
+function ParseTSV(contents) {
```
Comment (david-nishi) [inline bids-validator/validators/tsv/tsvParser.js]: Let's switch to camelCase for function and variable names.

File: bids-validator/validators/tsv/tsvParser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  for (let i = 0; i < Content.Rows.length; i++) {
+    var row = Content.Rows[i]
+    if (false) {
+      break
+    }
+    if (!row || /^\s*$/.test(row)) {
+      continue
+    }
+    Content.Values.push(row.trim().split('\t'))
+  }
```
Comment (david-nishi) [inline bids-validator/validators/tsv/tsvParser.js]: We should refactor this loop for clarity. There are some unused variables and and a leftover conditional statement.

File: bids-validator/validators/tsv/tsv.js [hunk 3]
```diff
@@ -26,27 +28,22 @@ const TSV = (file, contents, fileList, callback) => {
...
-    const row = rows[i]
-    if (columnMismatch && emptyCells && NACells) {
-      break
-    }
 
-    // skip empty rows
-    if (!row || /^\s*$/.test(row)) {
-      continue
+  rows.forEach((values, i) => {
+    var row = rows[i]
```
Comment (david-nishi) [inline bids-validator/validators/tsv/tsv.js]: row is used for evidence, so I think we should be setting this to something more informative. Maybe `row ${i}: ${values.join('\t')}`

**Review by david-nishi — CHANGES_REQUESTED** [review]

**Review by david-nishi — APPROVED** [review]

File: bids-validator/validators/tsv/tsvParser.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    headers: [],
+    rows: [],
+  }
+  content.rows = trimSplit('\n')(contents)
+    .filter(isContentfulRow)
+    .map(trimSplit('\t'))
+  content.headers = content.rows.length ? content.rows[0] : [] 
+  return content
+}
+module.exports = parseTSV
```
Comment (nellh) [inline bids-validator/validators/tsv/tsvParser.js]: This should be an ES6 export.

File: bids-validator/validators/bids/fullTest.js [hunk 5]
```diff
@@ -28,17 +28,17 @@ const fullTest = (fileList, options, annexed, dir, callback) => {
...
-    },
-    jsonFiles = [],
-    headers = [],
-    participants = null,
-    phenotypeParticipants = []
+    }
+  let jsonFiles = []
+  let headers = []
+  let participants = null
+  let phenotypeParticipants = []
```
Comment (nellh) [inline bids-validator/validators/bids/fullTest.js]: These should all be `const`.

File: bids-validator/validators/tsv/tsvParser.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+const trimSplit = separator => str => str.trim().split(separator)
+const isContentfulRow = row => row && !/^\s*$/.test(row)
+
+function parseTSV(contents) {
+  let content = {
+    headers: [],
+    rows: [],
+  }
```
Comment (nellh) [inline bids-validator/validators/tsv/tsvParser.js:13]: This should also be `const`.

**Review by nellh — CHANGES_REQUESTED** [review]
Took a quick look before merging this and there's still one issue I think should be fixed before merge, the mixed export styles should be ES6 exports. We've switched away from CJS in #864

**Pull request merged**